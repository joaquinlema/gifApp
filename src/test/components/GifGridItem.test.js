import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";
import '@testing-library/jest-dom';

describe('Pruebas sobre gif grid item', () => {

    const title = 'title test';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(
        <GifGridItem
            title={title}
            url={url}
        />
    );


    test('Comparamos contra snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Comprobamos que tenga P title', () => {

        const p = wrapper.find('p').text().trim();

        expect(p).toBe(title);
    });

    test('Comprobamos que tenga img title y alt', () => {

        const img = wrapper.find('img');

        expect(img.props().src.trim()).toBe(url);
        expect(img.props().alt.trim()).toBe(title);
    });

    test('Comprobamos que tenga div copn clase animate__fadeIn', () => {

        const div = wrapper.find('div');
        expect(div.props().className.includes('animate__fadeIn')).toBe(true);
    });

});
