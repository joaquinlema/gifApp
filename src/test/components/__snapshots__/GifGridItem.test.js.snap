// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre gif grid item Comparamos contra snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title test"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    title test
     
  </p>
</div>
`;
